src/main/java/me/jellysquid/mods/phosphor/mixin/chunk/MixinWorldChunk.java

```diff
@@ -34,25 +34,26 @@
      */
     @Overwrite
     public Stream<BlockPos> getLightSourcesStream() {
-        List<BlockPos> list = new ArrayList<>();
-
-        int startX = this.pos.getStartX();
-        int startZ = this.pos.getStartZ();
+        final int startX = this.pos.getStartX();
+        final int startZ = this.pos.getStartZ();
 
         ChunkSection[] chunkSections = this.sections;
 
+        // Pre-allocate array list for better performance
+        // `sections.length` does not affect the performance because an array has an fixed size
+        List<BlockPos> list = new ArrayList<>(sections.length * 16 * 16 * 16);
```

Comment:
How is this supposed to improve performance? There are generally very few light sources in a chunk (aside from a small amount of lava), and this just results in a ~500KB array always being allocated. Most of the array will never be used, and we will waste a lot of CPU/GC time on this large allocation.

Reply:
Well, this way you allocate memory beforehand instead of having to re-allocate it every time you add an element to a list. However, I agree with you that the GC has a lot to do there, I remove this implementation.